app/connectors/ReferenceDataConnector.scala

```diff
@@ -16,34 +16,83 @@
 
 package connectors
 
+import cats.Order
+import cats.data.NonEmptySet
 import config.FrontendAppConfig
 import connectors.ReferenceDataConnector.NoReferenceDataFoundException
 import logging.Logging
 import models.referenceData.{ControlType, CustomsOffice, FunctionalErrorWithDesc, RequestedDocumentType}
 import play.api.http.Status.OK
 import play.api.libs.json.{JsError, JsResultException, JsSuccess, Reads}
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpReads, HttpResponse}
+import sttp.model.HeaderNames
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, HttpResponse, StringContextOps}
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class ReferenceDataConnector @Inject() (config: FrontendAppConfig, http: HttpClient) extends Logging {
+class ReferenceDataConnector @Inject() (config: FrontendAppConfig, http: HttpClientV2) extends Logging {
 
   private type QueryParams = Seq[(String, String)]
 
-  private def version2Header: Seq[(String, String)] = Seq(
-    "Accept" -> "application/vnd.hmrc.2.0+json"
-  )
+  def getCustomsOffices(
+    queryParams: QueryParams
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[CustomsOffice]] = {
+    val url = url"${config.customsReferenceDataUrl}/lists/CustomsOffices"
+    http
+      .get(url)
+      .setHeader(version2Header)
+      .transform(_.withQueryStringParameters(queryParams: _*))
+      .execute[NonEmptySet[CustomsOffice]]
+  }
+
+  def getControlTypes(queryParams: QueryParams)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[ControlType]] = {
+    val url = url"${config.customsReferenceDataUrl}/lists/ControlType"
+    http
+      .get(url)
+      .setHeader(version2Header)
+      .transform(_.withQueryStringParameters(queryParams: _*))
+      .execute[NonEmptySet[ControlType]]
+  }
+
+  def getRequestedDocumentTypes(queryParams: QueryParams)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[RequestedDocumentType]] = {
+    val url = url"${config.customsReferenceDataUrl}/lists/RequestedDocumentType"
+    http
+      .get(url)
+      .setHeader(version2Header)
+      .transform(_.withQueryStringParameters(queryParams: _*))
+      .execute[NonEmptySet[RequestedDocumentType]]
+  }
+
+  def getFunctionalErrors(queryParams: QueryParams)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[FunctionalErrorWithDesc]] = {
+    val url = url"${config.customsReferenceDataUrl}/lists/FunctionalErrorCodesIeCA"
+    http
+      .get(url)
+      .setHeader(version2Header)
+      .transform(_.withQueryStringParameters(queryParams: _*))
+      .execute[NonEmptySet[FunctionalErrorWithDesc]]
+  }
+
+  def getFunctionalErrors()(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[NonEmptySet[FunctionalErrorWithDesc]] = {
```

Comment:
I suspect we can do away with the duplication here. Perhaps change `private type QueryParams = Seq[(String, String)]` to `private type QueryParam = (String, String)` and then in each method change `queryParams: QueryParams` to `queryParam: QueryParam*`